contracts/root/StakeManager.sol

```diff
@@ -199,29 +274,19 @@ contract StakeManager is StakeManagerInterface, RootChainable, Lockable {
     // total signers
     uint256 totalSigners = 0;
 
-    address lastAdd = address(0); // cannot have address(0) as an owner
+    address lastAdd = address(0x0); // cannot have address(0x0) as an owner
     for (uint64 i = 0; i < sigs.length; i += 65) {
       bytes memory sigElement = BytesLib.slice(sigs, i, 65);
       address signer = h.ecrecovery(sigElement);
 
       // check if signer is stacker and not proposer
-      if (totalStakedFor(signer) > 0 && signer != getProposer() && signer > lastAdd) {
+      if (totalStakedFor(signer) > 0 && signer != proposer && signer > lastAdd) {
         lastAdd = signer;
         totalSigners++;
       } else {
         break;
       }
     }
-
-    return totalSigners;
-  }
-
-  function validatorThreshold() public view returns (uint256) {
-    return _validatorThreshold;
-  }
-
-  function finalizeCommit(address proposer) public onlyRootChain {
-    // set epoch seed
-    epochSeed = keccak256(abi.encodePacked(block.difficulty + block.number + now));
+    return totalSigners >= validatorThreshold.mul(2).div(3).add(1);
```

Comment:
We should for totalStake - not count. @vaibhavchellani please confirm from tendermint

Reply:
we have to validate that signers have 2/3 of the total voting power .